packages/logs/src/boot/logsPublicApi.ts

```diff
@@ -125,6 +129,25 @@ export function makeLogsPublicApi(startLogsImpl: StartLogs) {
     getInitConfiguration: monitor(() => getInitConfigurationStrategy()),
 
     getInternalContext: monitor((startTime?: number | undefined) => getInternalContextStrategy(startTime)),
+
+    setUser: monitor((newUser: User) => {
+      if (typeof newUser !== 'object' || !newUser) {
+        display.error('Unsupported user:', newUser)
+      } else {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Maybe this check can also be mutualized in the `code/src/domain/user` module?